lib/database-ops.js

```diff
@@ -101,46 +101,64 @@ var changeUserPasswordInDatabase = function(userID, hashedPassword) {
 
 var startDatabaseClient = function(callback, options) {
     var dbURL = (options !== undefined && options.dbURL !== undefined) ? options.dbURL : url;
-    MongoClient.connect(dbURL, function(err, db) {
-        if (err) {
-            callback(err);
-            return;
-        }
+    MongoClient.connect(dbURL).then((db) => {
         logger.info("Successfully connected to database.");
         database = db;
 
-        var finishInitialization = function () {
-            imageEntryCollection = database.collection("image-entries");
+        var finishInitialization = function () {  
             commentsCollection = database.collection("comments");
             logCollection = database.collection("logs");
             actionHistoryCollection = database.collection("action-history");
 
             usersCollection.createIndex({ username: 1 }, { unique: true, sparse: true });
-            callback(err);
+            callback();
         };
 
-        database.listCollections().toArray(function(err, collections) {
-            if (err) {
-                throw err;
+        var CheckForImageEntriesCollection = function(collectionNames) {
+            if(collectionNames.indexOf("image-entries") < 0) {
+                database.createCollection("image-entries").then((collection) => {
+                    collection.createIndex({ uploadeddate: 1 }, { 
+                        name: "expire_index",
+                        expireAfterSeconds: 300,
```

Comment:
Perhaps we should make `expireAfterSeconds` configurable from environment variable?

Reply:
Yes, much better. I will be setting the default to 300.